src/arch/arm/machine/gic_v3.c

```diff
@@ -345,34 +340,19 @@ BOOT_CODE void cpu_initLocalIRQController(void)
 }
 
 #ifdef ENABLE_SMP_SUPPORT
-#define MPIDR_MT(x)   (x & BIT(24))
-
-void ipi_send_target(irq_t irq, word_t cpuTargetList)
+/* This function is called with a single node. */
+void ipi_send_target(irq_t irq, word_t cpuID)
```

Comment:
The prototype for this function is in `include/smp/ipi.h`, but the semantics change here. We have align this everywhere then to have a consistent behavior, otherwise this will get quite confusing across architectures.

Reply:
The semantics change here because the RISC-V (and I think x86) kernel has a similar example. Really, ipi_send_target is only used to send to one target at a time, and this could be reworked to be explicit instead.

However, I'd like to avoid touching any code that might require additional verification efforts for this port.

Reply:
Well, if we have a function with a feature set nobody uses, but everybody uses this in a simpler form, then it seems worth cleaning this up instead of making more wierd implementations. At least we could make dedicated PR for that, so the verification effort becomes better assessible. There are clear use cases why this change seems useful - or somebody can put up use cases we are not seeing.